tests/dataloader/test_metric.py

```diff
@@ -588,15 +542,17 @@ def test_add_metric(self):
 
 	def test_close1(self):
 		dataloader = FakeDataLoader()
-		data = dataloader.get_data('reference_key', 'reference_len_key', 'gen_prob_key',\
-								   'gen_key', 'post_key', 'resp_key', 'res_key', \
-								   True, False, True)
-		pm = MultiTurnPerplexityMetric(dataloader, 'reference_key', 'reference_len_key', 'gen_prob_key', full_check=True)
-		perplexity = TestMultiTurnPerplexityMetric().get_perplexity(data)
+		data = dataloader.get_data(reference_key='reference_key', reference_len_key='reference_len_key', gen_prob_key='gen_prob_key', \
+								   gen_key='gen_key', post_key='post_key', resp_key='resp_key', \
+								   multi_turn=True)
+		pm = MultiTurnPerplexityMetric(dataloader, 'reference_key', 'reference_len_key', 'gen_prob_key',
+									   full_check=True)
+		perplexity = TestMultiTurnPerplexityMetric().get_perplexity(data, 'reference_key', 'reference_len_key', 'gen_prob_key')
 
-		bcm = MultiTurnBleuCorpusMetric(dataloader, 'reference_key', 'res_key')
-		bleu = TestMultiTurnBleuCorpusMetric().get_bleu(dataloader, data)
+		bcm = MultiTurnBleuCorpusMetric(dataloader, 'reference_key', 'gen_key')
+		bleu = TestMultiTurnBleuCorpusMetric().get_bleu(dataloader, data, 'reference_key', 'gen_key')
 
+		_data = data
```

Comment:
python‘s assignment is reference assignment，you should copy data. See python library named `copy`.